src/status_im/contexts/wallet/add_account/create_account/utils.cljs

```diff
@@ -1,19 +1,29 @@
-(ns status-im.contexts.wallet.add-account.create-account.utils)
+(ns status-im.contexts.wallet.add-account.create-account.utils
+  (:require [status-im.constants :as constants]))
 
-(defn prepare-new-keypair
-  [{:keys [new-keypair address account-name account-color emoji derivation-path]}]
-  (assoc new-keypair
-         :name         (:keypair-name new-keypair)
-         :key-uid      (:keyUid new-keypair)
-         :type         :seed
-         :derived-from address
-         :accounts     [{:keypair-name (:keypair-name new-keypair)
-                         :key-uid      (:keyUid new-keypair)
-                         :seed-phrase  (:mnemonic new-keypair)
-                         :public-key   (:publicKey new-keypair)
-                         :name         account-name
-                         :type         :seed
-                         :emoji        emoji
-                         :colorID      account-color
-                         :path         derivation-path
-                         :address      (:address new-keypair)}]))
+(defn first-derived-account
+  [account-data]
+  (-> account-data :derived first val))
+
+(defn prepare-new-account
+  [{keypair-name                         :keypair-name
+    {:keys [keyUid address] :as account} :account-data
+    {:keys [account-name color emoji]}   :account-preferences}]
+  (let [account-to-create (first-derived-account account)
+        account-config    {:address    (:address account-to-create)
+                           :key-uid    keyUid
+                           :wallet     false
+                           :chat       false
+                           :type       :seed
+                           :path       constants/wallet-initial-derivation-path
+                           :public-key (:publicKey account-to-create)
+                           :name       account-name
+                           :emoji      emoji
+                           :colorID    color
+                           :hidden     false}]
+    {:key-uid                    keyUid
+     :name                       keypair-name
+     :type                       :seed
```

Comment:
For later - This (keypair type) should be received in the args as we have `seed` (recovery phrase), `key` (private key), and `profile`.

Reply:
Ok, I wasn't aware of the different types, I'll take it into consideration. Thank you!